test/test_custom_ops.py

```diff
@@ -2309,6 +2309,25 @@ def foo3(x: Tensor, *, y: int, z: List[Tensor]) -> Tensor:
                     setup_context=lambda ctx, inputs, keyword_only_inputs, output: None,
                 )
 
+            def foo_impl(x: torch.Tensor, *, y: torch.Tensor) -> torch.Tensor:
+                return x + y
+
+            lib.impl("foo", foo_impl, "CPU")
+
+            called = False
+
+            def myvmap(info, in_dims, x, *, y):
+                nonlocal called
+                called = True
+                return x + y, 0
+
+            torch.library.register_vmap("_torch_testing::foo", myvmap)
+            x = torch.randn(2, 3, requires_grad=True) * 2
+            y = torch.randn(3)
+            result = torch.vmap(torch.ops._torch_testing.foo)(x, y=y)
+            self.assertEqual(x + y, result)
+            self.assertTrue(called)
+
```

Comment:
I changed the kwarg_only test in test_custom_ops.py to reflect that we do support kwarg-only tensors. I can change it back if that's not right.

Reply:
Same question applies as in the autograd.Function case: what happens if the user writes the following?
```py
import torch
lib = torch.library.Library("mylib", "FRAGMENT")
lib.define("foo(Tensor x, *, Tensor y) -> Tensor")

def foo_impl(x, *, y):
    return x + y

lib.impl("foo", foo_impl, "CPU")

def myvmap(info, in_dims, x, *, y):
    return x + y, 0

def f(x, y):
    return torch.ops.mylib.foo(x, y=y)

torch.library.register_vmap("mylib::foo", myvmap)
x = torch.ones(2, 3, requires_grad=True) * 2
y = torch.ones(2, 3)
result = torch.vmap(f)(x, y)
```